functions/Get-DbaServerInstallDate.ps1

```diff
@@ -0,0 +1,181 @@
+function Get-DbaServerInstallDate
+{
+<#
+.SYNOPSIS
+Returns the install date of SqlServer and Windows Server, depending on what is passed. 
+	
+.DESCRIPTION
+By default, this command returns for each SQL Server instance passed in:
+SQL Instance install date, formatted as a string
+Hosting Windows server install date, formatted as a string
+	
+.PARAMETER SqlServer
+The SQL Server that you're connecting to.
+
+.PARAMETER SqlCredential
+Credential object used to connect to the SQL Server as a different user
+
+.PARAMETER WindowsCredential
+Credential object used to connect to the SQL Server as a different user
+
+.PARAMETER SqlOnly
+Excludes the Windows server information
+
+.PARAMETER WindowsOnly
+Excludes the SQL server information
+
+.PARAMETER Silent 
+Use this switch to disable any kind of verbose messages
+
+.NOTES
+Tags: CIM 
+Original Author: Mitchell Hamann (@SirCaptainMitch), mitchellhamann.com
+	
+Website: https://dbatools.io
+Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+.LINK
+https://dbatools.io/Get-DbaInstallDate
+
+.EXAMPLE
+Get-DbaInstallDate -SqlServer SqlBox1\Instance2
+
+Returns an object with SQL Server Install date as a string and the Windows install date as string. 
+
+.EXAMPLE
+Get-DbaInstallDate -SqlServer winserver\sqlexpress, sql2016
+
+Returns an object with SQL Server Install date as a string and the Windows install date as a string for both SQLInstances that are passed to the cmdlet.  
+	
+.EXAMPLE   
+Get-DbaInstallDate -SqlServer sqlserver2014a, sql2016 -SqlOnly
+
+Returns an object with only the SQL Server Install date as a string. 
+
+.EXAMPLE   
+Get-DbaInstallDate -SqlServer sqlserver2014a, sql2016 -WindowsOnly
+
+Returns an object with only the Windows Install date as a string. 
+
+.EXAMPLE   
+Get-SqlRegisteredServerName -SqlServer sql2014 | Get-DbaInstallDate
+
+Returns an object with SQL Server Install date as a string and the Windows install date as string for every server listed in the Central Management Server on sql2014
+	
+#>
+	[CmdletBinding(DefaultParameterSetName = "Default")]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlServer", "ComputerName")]
+		[DbaInstanceParameter[]]$SqlInstance,
+		[PSCredential] 
+		[System.Management.Automation.CredentialAttribute()]$SqlCredential,
+		[PSCredential] 
+		[System.Management.Automation.CredentialAttribute()]$Credential,
+		[parameter(ParameterSetName = "Sql")]
+		[Switch]$SqlOnly,
+		[parameter(ParameterSetName = "Windows")]
+		[Switch]$WindowsOnly,		
+		[switch]$Silent
+	)
+	
+	PROCESS
+	{
+		foreach ($instance in $SqlServer)
+		{
+			if ($instance.Gettype().FullName -eq [System.Management.Automation.PSCustomObject] )
```

Comment:
If it's a pscustomobject we take .sqlinstance property???
Wouldn't it make more sense to check if the object has a .sqlinstance property?